### PR TITLE
CXP-944: Create a Scope provider for Attribute Options

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml [hunk 1]
```diff
@@ -90,3 +90,7 @@ services:
             - '@oro_filter.filter_utility'
         tags:
             - { name: oro_filter.extension.orm_filter.filter, type: datagrid_attribute_family_filter }
+
+    Akeneo\Pim\Structure\Component\Security\AttirbuteOptionsScopeMapper:
+        tags:
+            - { name: pim_api.security.scope_mapper, entity: 'attribute_options' }
```
Comment (masacc) [inline src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml:96]:
```suggestion
            - { name: pim_api.security.scope_mapper }
```

File: src/Akeneo/Pim/Structure/Component/Security/AttributeOptionsScopeMapper.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    public function getAcls(string $scopeName): array
+    {
+        if (!\array_key_exists($scopeName, self::SCOPE_ACL_MAP)) {
+            return [];
+        }
+
+        return self::SCOPE_ACL_MAP[$scopeName];
+    }
+
+    public function getMessage(string $scope): array
```
Comment (masacc) [inline src/Akeneo/Pim/Structure/Component/Security/AttributeOptionsScopeMapper.php:63]: `$scope` => `$scopeName`

File: tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+namespace Specification\Akeneo\Pim\Structure\Component\Security;
+
+use Akeneo\Connectivity\Connection\Infrastructure\Apps\Security\ScopeMapperInterface;
+use Akeneo\Pim\Structure\Component\Security\AttributeOptionsScopeMapper;
+use PhpSpec\ObjectBehavior;
+
+class AttributeOptionsScopeMapperSpec extends ObjectBehavior
+{
+    public function it_is_a_attribute_options_scope_mapper(): void
```
Comment (masacc) [inline tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php:13]:
```suggestion
    public function it_is_an_attribute_options_scope_mapper(): void
```

File: tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            'pim_api_attribute_options_edit',
+        ]);
+    }
+
+    public function it_does_not_provide_acl_if_an_unknown_scope_is_given(): void
+    {
+        $this->getAcls('unknown_scope')->shouldReturn([]);
+    }
+
+    public function it_provides_message_that_correspond_to_read_attribute_options_scope(): void
```
Comment (masacc) [inline tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php:46]:
```suggestion
    public function it_provides_message_that_corresponds_to_the_read_attribute_options_scope(): void
```

File: tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $this->getMessage('read_attribute_options')->shouldReturn([
+            'icon' => 'attribute_options',
+            'type' => 'view',
+            'entities' => 'attribute_options',
+        ]);
+    }
+
+    public function it_provides_message_that_correspond_to_write_attribute_options_scope(): void
```
Comment (masacc) [inline tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php:55]:
```suggestion
    public function it_provides_message_that_corresponds_to_the_write_attribute_options_scope(): void
```

**Review by masacc — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml [hunk 6]
```diff
@@ -90,3 +90,7 @@ services:
             - '@oro_filter.filter_utility'
         tags:
             - { name: oro_filter.extension.orm_filter.filter, type: datagrid_attribute_family_filter }
+
+    Akeneo\Pim\Structure\Component\Security\AttirbuteOptionsScopeMapper:
```
Comment (masacc) [inline src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml:94]:
```suggestion
    Akeneo\Pim\Structure\Component\Security\AttributeOptionsScopeMapper:
```

Comment (guillaumefer-akeneo) [reply src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml]: fixed.

File: tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php (hunk 3 above)
Comment (guillaumefer-akeneo) [reply tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php]: fixed.

File: tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php (hunk 4 above)
Comment (guillaumefer-akeneo) [reply tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php]: fixed.

File: tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php (hunk 5 above)
Comment (guillaumefer-akeneo) [reply tests/back/Pim/Structure/Specification/Component/Security/AttributeOptionsScopeMapperSpec.php]: fixed.

File: src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml [hunk 7]
```diff
@@ -90,3 +90,7 @@ services:
             - '@oro_filter.filter_utility'
         tags:
             - { name: oro_filter.extension.orm_filter.filter, type: datagrid_attribute_family_filter }
+
+    Akeneo\Pim\Structure\Component\Security\AttributeOptionsScopeMapper:
```
Comment (RiffVentura) [inline src/Akeneo/Pim/Structure/Bundle/Resources/config/services.yml:94]: same comment : declaration onto a new file

**Review by RiffVentura — APPROVED** [review]

**Review by tseho — APPROVED** [review]

**Pull request merged**